File: R/R/measure.R
#' Measure mean squared error.
#' 
#' Given the ground truth (correct) target values and the estimated target 
#' values, calculates the mean squared error metric.
#'
#' @param y_true A numeric vector; the ground truth (correct) target values.
#' @param y_pred A numeric vector; the estimated target values.
#' @return A numeric vector of length one; the mean squared error metric.
#' @family measure
#' @export
measure_mse_score <- scorer::mean_squared_error

#' Measure Coefficient of Determination (R^2 Score).
#' 
#' Given the ground truth (correct) target values and the estimated target 
#' values, calculates the the R^2 metric.
#' 
#' See here for more information: \url{https://en.wikipedia.org/wiki/Coefficient_of_determination}
#'
#' @param y_true A numeric vector; the ground truth (correct) target values.
#' @param y_pred A numeric vector; the estimated target values.
#' @return A numeric vector of length one; the R^2 metric.
#' @family measure
#' @export
measure_r2_score <- function(y_true, y_pred) {
  measure_correlation_score(y_true, y_pred) ^ 2
}

#' Measure Pearsons Correlation Coefficient.
#' 
#' Given the ground truth (correct) target values and the estimated target 
#' values, calculates the correlation metric.
#' 
#' See here for more information: \url{https://en.wikipedia.org/wiki/Pearson_correlation_coefficient}
#'
#' @param y_true A numeric vector; the ground truth (correct) target values.
#' @param y_pred A numeric vector; the estimated target values.
#' @return A numeric vector of length one; the correlation metric.
#' @family measure
#' @export
measure_correlation_score <- function(y_true, y_pred) {
  stats::cor(y_true, y_pred)
}

#' Measure area under the curve.
#' 
#' Given the ground truth (correct) target values and the estimated target 
#' values, calculates the the AUC metric.
#'
#' @param y_true A numeric vector; the ground truth (correct) target values.
#' @param y_pred A numeric vector; the estimated target values.
#' @return A numeric vector of length one; the AUC metric.
#' @family measure
#' @export
measure_auc_score <- function(y_true, y_pred) {
  as.numeric(pROC::roc(as.numeric(y_true), as.numeric(y_pred))$auc)
}

File: Python/easymlpy/measure.py
"""
Functions for measuring model performance.
"""
import numpy as np
from sklearn.metrics import mean_squared_error, roc_auc_score


__all__ = ['measure_mean_squared_error', 'measure_cor_score',
           'measure_r2_score', 'measure_area_under_curve']


def measure_mean_squared_error(y_true, y_pred):
    """
    Measure mean squared error.

    Given the ground truth (correct) target values and the estimated target
    values, calculates the correlation metric.

    :param y_true: An ndarray; the ground truth (correct) target values.
    :param y_pred: An ndarray; the estimated target values.
    :return: A float.
    """
    return mean_squared_error(y_true, y_pred)


def measure_cor_score(y_true, y_pred):
    """
    Measure Pearsons Correlation Coefficient.

    Given the ground truth (correct) target values and the estimated target
    values, calculates the mean squared error metric.

    :param y_true: An ndarray; the ground truth (correct) target values.
    :param y_pred: An ndarray; the estimated target values.
    :return: A float.
    """
    return np.corrcoef(y_true, y_pred)[0, 1]


def measure_r2_score(y_true, y_pred):
    """
    Measure Coefficient of Determination (R^2 Score).

    Given the ground truth (correct) target values and the estimated target
    values, calculates the the R^2 metric.

    :param y_true: An ndarray; the ground truth (correct) target values.
    :param y_pred: An ndarray; the estimated target values.
    :return: A float.
    """
    return measure_cor_score(y_true, y_pred) ** 2


def measure_area_under_curve(y_true, y_pred):
    """
    Measure area under the curve.

    Given the ground truth (correct) target values and the estimated target
    values, calculates the the AUC metric.

    :param y_true: An ndarray; the ground truth (correct) target values.
    :param y_pred: An ndarray; the estimated target values.
    :return: A float.
    """
    return roc_auc_score(y_true, y_pred)
